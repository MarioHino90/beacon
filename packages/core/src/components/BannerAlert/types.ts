import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
