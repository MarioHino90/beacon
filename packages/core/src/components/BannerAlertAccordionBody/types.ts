export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
