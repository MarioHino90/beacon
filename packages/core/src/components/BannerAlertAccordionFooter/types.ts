export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
