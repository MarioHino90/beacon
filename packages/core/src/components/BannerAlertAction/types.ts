import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
