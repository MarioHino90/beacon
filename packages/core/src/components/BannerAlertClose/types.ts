import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
