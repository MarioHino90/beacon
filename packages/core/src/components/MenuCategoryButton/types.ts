import { BrazeIconProps } from "../BrazeIcon";
import { MenuButtonProps } from "../MenuButton";

export interface MenuCategoryButtonProps extends MenuButtonProps {
  /** heading to be displayed in the button */
  heading?: string;
  /** name of spot icon to be rendered, if any */
  iconName?: BrazeIconProps["name"];
}
