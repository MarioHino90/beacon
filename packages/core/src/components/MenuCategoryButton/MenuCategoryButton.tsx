import React from "react";
import classNames from "classnames";

import { BodyText } from "../BodyText";
import { Box } from "../Box";
import { BrazeIcon } from "../BrazeIcon";
import { Flex } from "../Flex";
import { FlexBox } from "../FlexBox";
import { Heading } from "../Heading";
import { MenuButton } from "../MenuButton";

import { MenuCategoryButtonProps } from "./types";

export const MenuCategoryButton = ({
  children,
  className,
  heading,
  iconName,
  ...props
}: MenuCategoryButtonProps): React.ReactElement => (
  <li role="none">
    <MenuButton
      {...props}
      className={classNames(["bcl-menu-category-button", className])}
      role="menuitem"
      tabIndex={-1}
      aria-label={
        typeof children === "string" ? children.toLowerCase() : undefined
      }
    >
      {iconName && (
        <FlexBox
          style={{ minWidth: 32 }}
          justifyContent="center"
          alignItems="center"
          boxMargin={{ right: "lg" }}
        >
          <BrazeIcon name={iconName} />
        </FlexBox>
      )}
      <Flex direction="column">
        <Box boxMargin={{ bottom: "xs" }}>
          <Heading level={5} variant="light">
            {heading}
          </Heading>
        </Box>
        <BodyText size="sm">{children}</BodyText>
      </Flex>
    </MenuButton>
  </li>
);

MenuCategoryButton.displayName = "MenuCategoryButton";
