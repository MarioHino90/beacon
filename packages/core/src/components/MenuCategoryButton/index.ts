export { MenuCategoryButton } from "./MenuCategoryButton";

export type { MenuCategoryButtonProps } from "./types";
