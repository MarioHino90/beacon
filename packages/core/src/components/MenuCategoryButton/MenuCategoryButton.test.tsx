import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { MenuCategoryButtonProps } from "./types";
import { MenuCategoryButton } from "./MenuCategoryButton";

faker.seed(1603821058478);

describe("MenuCategoryButton", () => {
  let props: MenuCategoryButtonProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<MenuCategoryButton {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when 'iconName' is defined", () => {
    const wrapper = shallow(
      <MenuCategoryButton {...props} iconName="campaign-comparison" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when 'heading' is defined", () => {
    const wrapper = shallow(
      <MenuCategoryButton {...props} heading={faker.random.words()} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when 'children' is NOT a string", () => {
    const wrapper = shallow(
      <MenuCategoryButton {...props}>
        <span>hi</span>
      </MenuCategoryButton>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
