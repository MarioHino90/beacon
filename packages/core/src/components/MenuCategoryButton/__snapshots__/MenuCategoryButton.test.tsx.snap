// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCategoryButton should match the snapshot 1`] = `
<li
  role="none"
>
  <MenuButton
    aria-label="compatible"
    className="bcl-menu-category-button Human"
    id="EXE"
    role="menuitem"
    tabIndex={-1}
  >
    <Flex
      direction="column"
    >
      <Box
        boxMargin={
          {
            "bottom": "xs",
          }
        }
      >
        <Heading
          level={5}
          variant="light"
        />
      </Box>
      <BodyText
        size="sm"
      >
        Compatible
      </BodyText>
    </Flex>
  </MenuButton>
</li>
`;

exports[`MenuCategoryButton should match the snapshot when 'children' is NOT a string 1`] = `
<li
  role="none"
>
  <MenuButton
    className="bcl-menu-category-button portals"
    id="niches"
    role="menuitem"
    tabIndex={-1}
  >
    <Flex
      direction="column"
    >
      <Box
        boxMargin={
          {
            "bottom": "xs",
          }
        }
      >
        <Heading
          level={5}
          variant="light"
        />
      </Box>
      <BodyText
        size="sm"
      >
        <span>
          hi
        </span>
      </BodyText>
    </Flex>
  </MenuButton>
</li>
`;

exports[`MenuCategoryButton should match the snapshot when 'heading' is defined 1`] = `
<li
  role="none"
>
  <MenuButton
    aria-label="malaysia personal licensed"
    className="bcl-menu-category-button violet"
    id="Future"
    role="menuitem"
    tabIndex={-1}
  >
    <Flex
      direction="column"
    >
      <Box
        boxMargin={
          {
            "bottom": "xs",
          }
        }
      >
        <Heading
          level={5}
          variant="light"
        >
          Assurance invoice Plastic
        </Heading>
      </Box>
      <BodyText
        size="sm"
      >
        Malaysia Personal Licensed
      </BodyText>
    </Flex>
  </MenuButton>
</li>
`;

exports[`MenuCategoryButton should match the snapshot when 'iconName' is defined 1`] = `
<li
  role="none"
>
  <MenuButton
    aria-label="utah payment wireless"
    className="bcl-menu-category-button dot-com"
    id="solution-oriented"
    role="menuitem"
    tabIndex={-1}
  >
    <FlexBox
      alignItems="center"
      boxMargin={
        {
          "right": "lg",
        }
      }
      justifyContent="center"
      style={
        {
          "minWidth": 32,
        }
      }
    >
      <BrazeIcon
        name="campaign-comparison"
      />
    </FlexBox>
    <Flex
      direction="column"
    >
      <Box
        boxMargin={
          {
            "bottom": "xs",
          }
        }
      >
        <Heading
          level={5}
          variant="light"
        />
      </Box>
      <BodyText
        size="sm"
      >
        Utah payment wireless
      </BodyText>
    </Flex>
  </MenuButton>
</li>
`;
